pandas/_libs/groupby.pyx

```diff
@@ -530,6 +531,9 @@ def group_add(add_t[:, ::1] out,
                     else:
                         t = sumx[lab, j] + val
                     sumx[lab, j] = t
+                elif skipna == False:
+                    # NOTE: Does this case need to be considered?
```

Comment:
yes.  if skipna is False and `not checknull(val)` (L524 above) then we `sumx[lab, j]` needs to be incremented by `val` (so will either become NaN or raise)